targets/csharp/source/source/PlayFabGameServerInternal.cs

```diff
@@ -0,0 +1,310 @@
+                }
+            }
+
+            if (!string.IsNullOrWhiteSpace(response.NextScheduledMaintenanceUtc))
+            {
+                if (DateTime.TryParse(
+                    response.NextScheduledMaintenanceUtc,
+                    null,
+                    DateTimeStyles.RoundtripKind,
+                    out DateTime scheduledMaintDate))
+                {
+                    if (_cachedScheduleMaintDate == DateTime.MinValue || (scheduledMaintDate != _cachedScheduleMaintDate))
+                    {
+                        MaintenanceCallback?.Invoke(scheduledMaintDate);
+                        _cachedScheduleMaintDate = scheduledMaintDate;
+                    }
+                }
+            }
+
+            switch (response.Operation)
+            {
+                case GameOperation.Continue:
+                    {
+                        // No action required
+                        break;
+                    }
+                case GameOperation.Active:
+                    {
+                        if (State != GameState.Active)
+                        {
+                            State = GameState.Active;
+                        }
+
+                        TransitionToActiveEvent.Set();
+
+                        break;
+                    }
+                case GameOperation.Terminate:
+                    {
+                        if (State != GameState.Terminating)
+                        {
+                            State = GameState.Terminating;
+                            ShutdownCallback?.Invoke();
+                        }
+
+                        TransitionToActiveEvent.Set();
+
+                        break;
+                    }
+                default:
+                    {
+                        Logger.Log($"Unknown operation received: {Enum.GetName(typeof(GameOperation), response.Operation)}");
+                        break;
+                    }
+
+            }
+
+            return Task.CompletedTask;
+        }
+
+#region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        private void DisposeManagedResources(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    _heartbeatRunning = false;
+                    _heartbeatDoneEvent.WaitOne();
+                }
+
+                // TODO: free unmanaged resources (unmanaged objects) and override a finalizer below.
+                // TODO: set large fields to null.
+
+                disposedValue = true;
+            }
+        }
+
+        // TODO: override a finalizer only if Dispose(bool disposing) above has code to free unmanaged resources.
+        // ~InternalSdk() {
+        //   // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+        //   Dispose(false);
+        // }
```

Comment:
Please don't submit commented code.
It looks like this should just be deleted.

Reply:
agreed, I'll just delete it and clean this up

---
In reply to: [256125381](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r256125381) [](ancestors = 256125381)